x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ldap/ActiveDirectoryGroupsResolver.java

```diff
@@ -53,16 +52,16 @@ public void resolve(LDAPInterface connection, String userDn, TimeValue timeout,
         buildGroupQuery(connection, userDn, timeout,
                 ignoreReferralErrors, ActionListener.wrap((filter) -> {
                     if (filter == null) {
-                        listener.onResponse(Collections.emptyList());
+                        listener.onResponse(List.of());
```

Comment:
Per my comment above, I think we need to decide whether we want to switch these across the board. I don't see any reason to do so, but I'm not specifically opposed to it.

Reply:
Per my comment above, the only reason is to get an `AbstractImmutableCollection` instead of an `AbstractCollection`, the former avoiding a new instance creation in the `Set.copyOf` or `List.copyOf` calls.